myapp/app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,58 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    @labels = @task.labels
+  end
+
+  def new
+    @task = Task.new
+    @labels = Label.all
+    @task_form_url = tasks_path
```

Comment:
where use this variable at new && edit?

@task_form_url
@task_form_method
@task_submit_label

Reply:
I've refactored the common parts of `edit.html.erb` and `new.html.erb` into a shared partial called `_form.html.erb`. This allows me to reuse the code for both actions while accommodating their differences in URL, HTTP request method, and submit button label.

By defining variables in the controller, I can pass the necessary values to the shared partial when rendering the views. This ensures that each form is properly configured for its specific action.

Reply:
Maybe you move it up outside the [form](https://github.com/Fablic/training/pull/1578/files#diff-228b16ec9fa0b8dfeef7e828c836e81b5b85a54bb690351188048c71d35511e3R1) from `_form.html.erb`
you don't need to for these variable

Reply:
Sure, I have move those into `edit.html.erb` and `new.html.erb`, and remove the reductand variables from task controller.